### PR TITLE
Tutorial: Unit testing stack references in go

### PR DESCRIPTION
**Pull request by desteves**
## Checklist:

- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [x] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

File: themes/default/content/docs/using-pulumi/testing/unit-sr.md [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+Pulumi programs are authored in general-purpose languages like TypeScript, Python, Go, C#, or Java. The full power of each language is available, including access to tools, libraries, and testing frameworks.
+
+When running an update, your Pulumi program talks to the Pulumi CLI to orchestrate any deployment changes. Unit tests aim to cut this communication channel and replace the engine with mocks. Mocks respond to the commands from _within_ the same process and return mock (or dummy) data for each call that your Pulumi program makes.
+
+Because mocks don't execute actual work, unit tests run very fast. Also, they can be deterministic because tests don't depend on the behavior of any external system.
+
+In Pulumi, stack references are a way to reference and use the outputs of one stack in another. Stack references are a common way to decouple infrastructure, handle cross-environment deployments, and manage dependencies. DevOps engineers may need to mock a `StackReference` as part of a CI/CD pipeline and to meet code coverage standards.
+
+[**Learn more about Stack References**](/docs/using-pulumi/testing/unit).
```
Comment (scottslowe) [inline themes/default/content/docs/using-pulumi/testing/unit-sr.md:24]: I think this link may be incorrect...?

**Review by scottslowe — CHANGES_REQUESTED** [review]
Overall, this is good. I got a bit lost in the middle of the process, so depending on the intended target audience, this tutorial might benefit from some slight additional context/information around each step and what it's doing.

File: themes/default/content/docs/using-pulumi/testing/unit-sr.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+#### b. Create the program
+
+```bash
+$ pulumi new https://github.com/pulumi/tutorials/tree/testing-unit-go-stackreferences-begin
+```
+
+### 2. Add the testing framework
+
+You can write unit tests and assertions using your favorite frameworks and libraries. We use Ginkgo and Gomega.
```
[the remark this answers is not in the document]
Comment (desteves) [reply themes/default/content/docs/using-pulumi/testing/unit-sr.md]: Done

**Comment by desteves** [comment]
> Overall, this is good. I got a bit lost in the middle of the process, so depending on the intended target audience, this tutorial might benefit from some slight additional context/information around each step and what it's doing.

Thanks for the feedback, Scott; I'll add info bubbles and more details for the concepts that fall more on the advanced end.

**Pull request closed without merging**